### PR TITLE
WT-10457 Data format change for page stat

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/include/btmem.h [hunk 1]
```diff
@@ -72,15 +72,14 @@ struct __wt_page_header {
...
-#define WT_PAGE_UNUSED 0x10u          /* Historic lookaside store page updates, no longer used */
-#define WT_PAGE_FT_UPDATE 0x20u       /* Page contains updated fast-truncate information */
-#define WT_PAGE_STAT_BYTE_COUNT 0x40u /* Page contains byte count information */
-#define WT_PAGE_STAT_ROW_COUNT 0x80u  /* Page contains row count information */
-    uint8_t flags;                    /* 25: flags */
+#define WT_PAGE_COMPRESSED 0x01u   /* Page is compressed on disk */
+#define WT_PAGE_EMPTY_V_ALL 0x02u  /* Page has all zero-length values */
+#define WT_PAGE_EMPTY_V_NONE 0x04u /* Page has no zero-length values */
+#define WT_PAGE_ENCRYPTED 0x08u    /* Page is encrypted on disk */
+#define WT_PAGE_UNUSED 0x10u       /* Historic lookaside store page updates, no longer used */
```
Comment (quchenhao) [inline src/include/btmem.h:79]: May be we can reuse this unused bit?

File: src/include/btmem.h [hunk 2]
```diff
@@ -72,15 +72,14 @@ struct __wt_page_header {
...
+#define WT_PAGE_FT_UPDATE 0x20u    /* Page contains updated fast-truncate information */
+#define WT_PAGE_SUPPORT_STAT 0x40u /* Page supports page stat */
+    uint8_t flags;                 /* 25: flags */
 
     /* A byte of padding, positioned to be added to the flags. */
     uint8_t unused; /* 26: unused padding */
```
Comment (quchenhao) [inline src/include/btmem.h:85]: Another choice is to use the version value instead of flags for page with stat information. @agorrod

Comment (quchenhao) [reply src/include/btmem.h:85]: Or should we set the flag but also bump the version since now we have page deleted structure and page stat?

Comment (quchenhao) [reply src/include/btmem.h:85]: I'm a bit confused why we need this version field. Seems it doesn't serve any functional purpose just a symbol of version.

File: src/include/btmem.h (hunk 1 above)
Comment (agorrod) [reply src/include/btmem.h:79]: These are durable flags (i.e: present in on-disk data files). The lookaside is special, in that it should not be present in a current release, so this flag *might* be able to be reused, but I'd rather not until/unless we need to.

**Review by agorrod — COMMENTED** [review]
@quchenhao - this PR is useful to see what the change would look like, but given that it's a data format change, I think it's important that we finalize *what* data format change we are going with and flush out all upgrade/downgrade consequences prior to approving this PR.

**Pull request closed without merging**